mixer/pkg/protobuf/yaml/castUtil.go

```diff
@@ -45,6 +47,14 @@ func ToInt64(v interface{}) (int64, bool) {
 		return int64(c), true
 	case int64:
 		return c, true
+	case float64:
+		// json only supports 'number'
+		// therefore even integers are encoded as float
+		ii := int64(c)
```

Comment:
This seems odd. We are trying to covert to float64 and the number is already that. I think this was added to the wrong place.

Reply:
We want the thing `c` to be a int64, but it comes to us as float64. This code is trying to cast it into int64 and see if we can do it without loosing data. This is similar to your code in `encoderUtil.go`, right ?

Reply:
Diff is hiding the location of the code. PTAL https://github.com/guptasu/istio/blob/5c200b33f1af2553f30d25c1ae78bb76390814ad/mixer/pkg/protobuf/yaml/castUtil.go